fix(styles): popover horizon and v1 refactoring

platon-rov opened the pull request:
## Related Issue

Refers to 

## Description

* Popover styles refactoring
* Popover docs refactoring
* Popover placement feature
* Popover minor issues fixes

*Only one misalignment with specs is the arrow position, it should be `0.5rem` accordingly to the designs but in that case arrow won't be centered to the button with icon.*

![CleanShot 2022-11-16 at 10 38 36@2x](https://user-images.githubusercontent.com/20265336/202145140-4bad1029-c4d9-4471-9261-604e706022ba.png)

## Screenshots

### Before:

<img width="359" alt="CleanShot 2022-11-15 at 22 25 20@2x" src="https://user-images.githubusercontent.com/20265336/202028441-b61ba35e-7385-4cae-8c55-293f1b477cb9.png">

### After:

<img width="347" alt="CleanShot 2022-11-15 at 22 25 43@2x" src="https://user-images.githubusercontent.com/20265336/202028503-4b5efc1a-2183-49af-b856-256eb655b0aa.png">

<img width="785" alt="CleanShot 2022-11-15 at 22 26 04@2x" src="https://user-images.githubusercontent.com/20265336/202028565-fafb575c-01e9-47e4-947c-68f9a097782b.png">

N1XUS reviewed (approved):
Nice job!

InnaAtanasova commented:
![Screenshot 2022-11-16 at 9 51 06 AM](https://user-images.githubusercontent.com/39598672/202213310-8d73d421-870d-4ea6-a440-bb5819d5fd92.png)

Hey @platon-rov please add the project and milestone to your PRs. It helps us to monitor what gets done during a sprint and also I don't have to go and add these to the PRs as a primary school teacher :D

InnaAtanasova commented:
@platon-rov I see changes in the markup (adding classes). This should be considered a Breaking Change, shouldn't it?

platon-rov commented:
> @platon-rov I see changes in the markup (adding classes). This should be considered a Breaking Change, shouldn't it?

Interesting question, accordingly to the semver.org any new feature or deprecation should bump minor (0.X.0) version number and in our case minor is a major because the major is always 0 because we're in beta. But we don't mark new features as breaking changes, because they don't break anything for sure. The same with deprecation, it doesn't brake anything by itself, only when we then remove this deprecation we introduce breaking changes, it makes sense. Having breaking changes twice (when deprecating (1) and then when removing (2)) for the same thing is a bit too much.

So in this exact case i deprecated some stuff and updated examples to don't use show deprecated things there.

g-cheishvili commented on packages/styles/stories/Components/popover/popover.stories.js:
```diff
@@ -34,176 +34,6 @@ export default {
     }
 };
 
-export const Alignment = () => `<div class="fddocs-container" style="margin-bottom: 200px">
```
FYI, You can extract HTML files into the separate HTML files if you want to for proper HTML formating or something

g-cheishvili commented:
> > @platon-rov I see changes in the markup (adding classes). This should be considered a Breaking Change, shouldn't it?
> 
> Interesting question, accordingly to the semver.org any new feature or deprecation should bump minor (0.X.0) version number and in our case minor is a major because the major is always 0 because we're in beta. But we don't mark new features as breaking changes, because they don't break anything for sure. The same with deprecation, it doesn't brake anything by itself, only when we then remove this deprecation we introduce breaking changes, it makes sense. Having breaking changes twice (when deprecating (1) and then when removing (2)) for the same thing is a bit too much.
> 
> So in this exact case i deprecated some stuff and updated examples to don't use show deprecated things there.

I get your point, related to double breaking changes, but what about `fd-popover--input-message-group` class relocations?

g-cheishvili reviewed (approved):
Accept new base on chromatic

platon-rov commented:
> I get your point, related to double breaking changes, but what about `fd-popover--input-message-group` class relocations?

I've added a new class `fd-popover__body--input-message-group` and changed usages in docs to use this way, but also left the old one `fd-popover--input-message-group` so it's still just a deprecation, not removing.
